### ISSUE TITLE
Support the new ZoneId field on Availability Zones

### ISSUE THREAD

**Issue by tecnobrat** [terraform-providers/terraform-provider-aws#6546]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The ec2 describe-availability-zones call now returns a ZoneId field which gives you the exact physical location of where that AZ is.  For example, the cli now returns:

```
            "State": "available",
            "ZoneName": "us-east-1a",
            "Messages": [],
            "ZoneId": "use1-az1",
            "RegionName": "us-east-1"
```

This is powerful for being able to exclude certain availability zones when they don't have supported instance types or features (EKS is one)

### New or Affected Resource(s)

data.aws_availability_zone
data.aws_availability_zones

Probably others.

**Comment by ewbankkit** [comment]
Note that `AvailabilityZoneId` has also now been added to the `CreateSubnet` API.

**Comment by bflad** [comment]
Support for zone ID in the two availability zone data sources has been merged and will release with version 1.51.0 of the AWS provider, likely in the next day or two. 👍
